ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,239 @@
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Bool
+        , "enabled"
+        , "If the Tracer Provider is to be Enabled (true) or disabled (false)"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the tags field of a tracer provider that are emitted to spans \
+       derived from this provider"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags that are emitted to spans derived from this provider"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_endpoints =
+  call ~name:"set_endpoints" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the file/HTTP endpoints the provider sends traces to"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Set String
+        , "endpoints"
+        , "The endpoints that the tracer provider will export spans to. A URL \
+           or the string 'bugtool'. This can refer to an enpoint to the local \
+           file system"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_components =
+  call ~name:"set_components" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the components on which the tracer provider is to registered on. \
+       i.e. xapi, xenopds, networkd, etc"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Set String
+        , "components"
+        , "The components the tracer provider is to be registered to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_filters =
+  call ~name:"set_filters" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the filters field of a tracer provider which will dictate what \
+       operations are to be traced"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; (Set String, "filters", "The filter the tracer provider is to be set to")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_processors =
+  call ~name:"set_processors" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the processors field of a tracer provider that define what \
+       operations need to take place prior to export"
+    ~params:
+      [
+        (Ref _tracing, "self", "The tracer provider")
+      ; ( Set String
+        , "processors"
+        , "The processors the tracer provider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let t =
+  create_obj ~name:_tracing ~descr:"Describes a tracer provider" ~doccomments:[]
+    ~gen_constructor_destructor:false ~gen_events:true ~in_db:true ~lifecycle:[]
+    ~persist:PersistEverything ~in_oss_since:None
+    ~messages_default_allowed_roles:_R_POOL_ADMIN
+    ~contents:
+      [
+        uid _tracing ~lifecycle:[]
+      ; field ~qualifier:DynamicRO ~ty:(Set (Ref _host)) ~lifecycle:[] "hosts"
```

Comment:
@edwintorok: I think you commented at some point that in general we would like to observe all hosts, but in some debugging conditions it would be useful to be able to enable the observer provider for individual hosts, is that the case?

Reply:
It would seem we have agreed to keep this